### PR TITLE
feat: One off campaign UI

### PR DESCRIPTION
**Pull request by muhsin-k**
- Added UI for creating a new SMS campaign
- Update campaign table to show SMS campaigns too
- Update the date picker for supporting date range and date-time

**Preview**

Screenshot 2021-07-14 at 7 03 44 PM

Screenshot 2021-07-14 at 7 04 13 PM

### CODE REVIEW

File: app/javascript/dashboard/components/ui/DatePicker.vue [hunk 1]
```diff
@@ -28,25 +30,51 @@ export default {
...
+  methods: {
     handleChange(value) {
-      this.updateValue(value);
+      this.$emit('change', value);
+    },
+    disableBeforeToday(date) {
+      if (!this.isTypeDateRange) {
+        const today = new Date();
+        const yesterday = new Date(today);
+        yesterday.setDate(yesterday.getDate() - 1);
```
Comment (nithindavid) [inline app/javascript/dashboard/components/ui/DatePicker.vue:58]: https://date-fns.org/v2.22.1/docs/addDays

File: app/javascript/dashboard/components/ui/DatePicker.vue [hunk 2]
```diff
@@ -1,13 +1,15 @@
 <template>
-  <div class="woot-date-range-picker">
+  <div class="date-picker">
     <date-picker
-      :range="true"
+      :type="type"
```
Comment (nithindavid) [inline app/javascript/dashboard/components/ui/DatePicker.vue:4]: Can we keep date range and date picker as separate components to keep it simple.

Comment (nithindavid) [reply app/javascript/dashboard/components/ui/DatePicker.vue:4]: If a future case comes up, we have to always check the type is Array or date depending on functionality.

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue [hunk 3]
```diff
@@ -166,9 +231,13 @@ export default {
     onClose() {
       this.$emit('on-close');
     },
+    onChange(value) {
+      this.scheduledAt = value;
+    },
     async addCampaign() {
-      try {
-        await this.$store.dispatch('campaigns/create', {
+      let campaignDetails = null;
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue:238]: Moving the `campaignDetails` logic to a new method/computed would be good.

**Review by nithindavid — CHANGES_REQUESTED** [review]

**Comment by nithindavid** [comment]
NIce work :) Added some minor comments.

**Review by nithindavid — APPROVED** [review]

**Comment by muhsin-k** [comment]
@nithindavid Please check the latest changes

**Comment by tabrza** [comment]
would be worth having the ability to link to the website wherever the chat widget is integrated and have it pre-opened in order to encourage engaging in a new conversation. this could be an option should the campaign creator wish so

**Pull request merged**